@osqp/setup.m

```diff
@@ -0,0 +1,106 @@
+%%
+function varargout = setup(this, varargin)
+    % SETUP configure solver with problem data
+    %
+    %   setup(P,q,A,l,u,options)
+
+    nargin = length(varargin);
+
+    %dimension checks on user data. Mex function does not
+    %perform any checks on inputs, so check everything here
+    assert(nargin >= 5, 'incorrect number of inputs');
+    [P,q,A,l,u] = deal(varargin{1:5});
+
+    %
+    % Get problem dimensions
+    %
+
+    % Get number of variables n
+    if (isempty(P))
+        if (~isempty(q))
+            n = length(q);
+        else
+            if (~isempty(A))
+                n = size(A, 2);
+            else
+                error('The problem does not have any variables');
+            end
+        end
+    else
+        n = size(P, 1);
+    end
```

Comment:
Yea, this block is awkward. a switch/case isn't appropriate though, since it isn't changing on values but instead based on what is empty/non-empty. I've refactored this into an if-elseif-else chain to be clearer.